initialisation.php

```diff
@@ -0,0 +1,170 @@
+
+    return $query->execute();
+}
+
+function createMainTable(Database $db): bool
+{
+    $sql = 'CREATE TABLE `cars` ('
+        . '`id` int(11) unsigned NOT NULL AUTO_INCREMENT,'
+        . '`make` int(11) unsigned DEFAULT NULL,'
+        . '`model` varchar(255) DEFAULT NULL,'
+        . '`year` int(11) DEFAULT NULL,'
+        . '`colour` int(11) unsigned DEFAULT NULL,'
+        . '`location` int(11) unsigned DEFAULT NULL,'
+        . '`image` varchar(255) DEFAULT NULL,'
+        . 'PRIMARY KEY (`id`),'
+        . 'CONSTRAINT `fk_cars_make` FOREIGN KEY (`make`) REFERENCES `makes` (`id`),'
+        . 'CONSTRAINT `fk_cars_location` FOREIGN KEY (`location`) REFERENCES `locations` (`id`),'
+        . 'CONSTRAINT `fk_cars_colour` FOREIGN KEY (`colour`) REFERENCES `colours` (`id`)'
+        . ') ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=latin1;';
+
+    $query = $db->getConnection()->prepare($sql);
+
+    return $query->execute();
+}
+
+// Fill tables
+function fillMainTable (Database $db, array $cars, array $reducedMakes, array $reducedColours, array $reducedLocations): bool
+{
+    $makeID = [];
+    $colourID = [];
+    $locationID = [];
+
+    foreach ($cars as $car) {
+
+        foreach ($reducedMakes as $reducedMake) {
+            if ($car['make'] == $reducedMake) {
+                $makeID = array_keys($reducedMakes, $reducedMake);
+                $makeID[0]++;
+            }
+        }
+
+        foreach ($reducedColours as $reducedColour) {
+            if (!$car['color']) {
+                $colourID[0] = 15;
+            } elseif ($car['color'] == $reducedColour) {
+                $colourID = array_keys($reducedColours, $reducedColour);
+                $colourID[0]++;
+            }
+        }
+
+        foreach ($reducedLocations as $reducedLocation) {
+            if ($car['location'] == $reducedLocation) {
+                $locationID = array_keys($reducedLocations, $reducedLocation);
+                $locationID[0]++;
+            }
+        }
+
+        if (!$car['year']) {
+            $car['year'] = null;
+        }
+
+        $sql = "INSERT INTO `cars` (`make`, `model`, `year`, `colour`, `location`, `image`)"
+            . " VALUES (:make, :model, :year, :colour, :location, :image); ";
+
+        $values = [
+            ':make' => $makeID[0],
+            ':model' => $car['model'],
+            ':year' => $car['year'],
+            ':colour' => $colourID[0],
+            ':location' => $locationID[0],
+            ':image' => $car['image']
+        ];
+
+        $query = $db->getConnection()->prepare($sql);
+
+        $query->execute($values);
+    }
+
+    return true;
+}
+
+function fillTables(Database $db, string $tableName, string $fieldName, array $data): bool
+{
+    foreach ($data as $dt) {
+        $sql = "INSERT INTO `$tableName` (`$fieldName`) VALUES (:data); ";
+
+        $values = [':data' => $dt];
+
+        $query = $db->getConnection()->prepare($sql);
+
+        $query->execute($values);
```

Comment:
As execute() returns a boolean:
If it fails, why not return the false so the calling code can display a message that the database creation failed?
```
$success = $query->execute($values);
if (!$success) {
    return false;
}
```